classes/audit_manager.php

```diff
@@ -0,0 +1,376 @@
+        $activity->courseurl = self::get_course_url_from_record($record);
+        $activity->coursefullname = $record->coursefullname;
+        $activity->type = str_replace('mod_', '', $record->component);
+        $activity->activityurl = self::get_activity_url_from_record($activity->type, $record);
+        $activity->activityname = $record->name;
+        $activity->html5present = empty($record->html5) ? 0 : 1;
+
+        return new audit_activity($activity);
+    }
+
+    /**
+     * Get category from record
+     *
+     * @param mixed $record a fieldset object containing a record
+     *
+     * @return string  The category name or empty string if none found
+     * @throws \dml_exception
+     */
+    public static function get_category_from_record($record) {
+        global $DB;
+
+        // Get the course category names and their ids.
+        $coursecategorynames = $DB->get_records_menu('course_categories', array(), '', 'id, name');
+        $category = '';
+        if (!empty($record->category)) {
+            $categories = explode('/', $record->category);
+            foreach ($categories as $c) {
+                if (!empty($c) && !empty($coursecategorynames[$c])) {
+                    if (!empty($category)) {
+                        $category .= " / ";
+                    }
+                    $category .= $coursecategorynames[$c];
+                }
+            }
+        }
+        return $category;
+    }
+
+
+    /**
+     * Get a course url representation from a db record.
+     *
+     * @param mixed $record a fieldset object containing a record
+     *
+     * @return string  Resulting URL
+     * @throws \moodle_exception
+     */
+    public static function get_course_url_from_record($record) {
+
+        $courseurl = new moodle_url('/course/view.php', array('id' => $record->id));
+        return $courseurl->out();
+
+    }
+
+    /**
+     * Get an activity url representation from a db record.
+     *
+     * @param string $type The type of the activity
+     * @param mixed $record a fieldset object containing a record
+     *
+     * @return string Resulting URL
+     * @throws \moodle_exception
+     */
+    public static function get_activity_url_from_record($type = '', $record) {
+        if ($type == 'legacy') {
+            $activityurl = new moodle_url("/files/index.php", array('contextid' => $record->contextid));
+        } else {
+            $activityurl = new moodle_url("/mod/$type/view.php", array('id' => $record->instanceid));
+        }
+        return $activityurl->out();
+    }
+
+    /**
+     * Insert audit date/time and a count of flash activities detected into 'report_apocalypse_audit' table.
+     *
+     * @param int $count the count of activities to store in record
+     *
+     * @throws \Exception
+     */
+    public static function insert_audit_record(int $count) {
+        global $DB;
+
+        $record = new stdClass();
+        $date = new DateTime();
```

Comment:
this seems weird - probably should just be:
$record->rundatetime = time();

Reply:
Mistake on my behalf, I didn't realise time() returned a Unix Epoch, I'll fix

Reply:
Fixed, removed superfluous code.